src/Search.php

```diff
@@ -5165,7 +5165,7 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
             case "glpi_ticketvalidations.status":
             case "glpi_changes.global_validation":
             case "glpi_changevalidations.status":
-                if ($val == 'all') {
+                if (!in_array($val, ['all', 'can'], true) && !is_numeric($val)) {
```

Comment:
In fact, I guess this is legitimate to apply no filter when the value is "all". It may improve performances and prevent unexpected filtering of "bad" values.

The "all" case could be removed from the switch.

```suggestion
                if ($val === 'all' || ($val !== 'can' && !is_numeric($val))) {
```